### PR TITLE
draft api for cloud data sources

### PR DESCRIPTION
**Pull request by ssoroka**

### CODE REVIEW

File: src/svc-cloud-data-sources.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+openapi: 3.1.0
+info:
+  title: Cloud Data Sources Service
+  version: '1.0'
+servers:
+  - url: /
+paths:
+  '/orgs/{orgID}/datasources':
```
Comment (glinton) [inline src/svc-cloud-data-sources.yml:8]: I believe we are wanting to move away from having the orgid in the route if possible. We're currently deriving the orgId from the token/session but i think there is a recent service that is still accepting orgID as a query parameter too.

File: src/svc-cloud-data-sources.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      $ref: ./svc/cloud-data-sources/schemas/Datasource.yml
+    ListenerEndpoint:
+      $ref: ./svc/cloud-data-sources/schemas/ListenerEndpoint.yml
+    Service:
+      $ref: ./svc/cloud-data-sources/schemas/Service.yml
+  responses:
+    NoContent:
+      $ref: ./common/responses/NoContent.yml
+    ServerError:
+      $ref: ./common/responses/ServerError.yml
```
Comment (glinton) [inline src/svc-cloud-data-sources.yml:24]: Are these responses going to be used by any `paths`?

File: src/svc/cloud-data-sources/paths/orgs-orgID-datasources.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  id:
+                    type: string
+                  endpoint:
+                    $ref: '../schemas/ListenerEndpoint'
```
Comment (glinton) [inline src/svc/cloud-data-sources/paths/orgs-orgID-datasources.yml:28]: This file has an extension
```suggestion
                    $ref: '../schemas/ListenerEndpoint.yml'
```

File: src/svc/cloud-data-sources/paths/orgs-orgID-datasources.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      operationId: get-orgs-orgID-datasources
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: ./svc/cloud-data-sources/schemas/Datasource.yml
```
Comment (glinton) [inline src/svc/cloud-data-sources/paths/orgs-orgID-datasources.yml:12]: these are relative
```suggestion
                  $ref: ../schemas/Datasource.yml
```

File: src/svc/cloud-data-sources/paths/orgs-orgID-datasources.yml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+              schema:
+                type: object
+                properties:
+                  id:
+                    type: string
+                  endpoint:
+                    $ref: '../schemas/ListenerEndpoint'
+      description: Create a datasource with a specific plugin configuration
+      requestBody:
+        description: ''
```
Comment (glinton) [inline src/svc/cloud-data-sources/paths/orgs-orgID-datasources.yml:31]: nit: remove keys with empty values

File: src/svc/cloud-data-sources/paths/orgs-orgID-datasources.yml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                name:
+                  type: string
+                config:
+                  type: object
```
Comment (glinton) [inline src/svc/cloud-data-sources/paths/orgs-orgID-datasources.yml:40]: does this object type have a specific definition?

File: src/svc-cloud-data-sources.yml [hunk 7]
```diff
@@ -0,0 +1,24 @@
+openapi: 3.1.0
```
Comment (glinton) [inline src/svc-cloud-data-sources.yml:1]: the swagger parser we are using only supports up to 3.0.3, is there a specific 3.1.0 swagger feature you're using?

**Review by glinton — COMMENTED** [review]
Fixing the api version should allow the contract to be properly generated when running `make generate-svc`.

File: src/svc/cloud-data-sources/paths/orgs-orgID-datasources.yml (hunk 6 above)
Comment (ssoroka) [reply src/svc/cloud-data-sources/paths/orgs-orgID-datasources.yml:40]: yes, but it's dynamic and I would think outside the scope of the api definition. basically the sum of all possible configs for all (supported) telegraf plugins (as json)

File: src/svc-cloud-data-sources.yml (hunk 7 above)
Comment (ssoroka) [reply src/svc-cloud-data-sources.yml:1]: I have no idea. :D

**Review by glinton — COMMENTED** [review]
merge master in and re-run `make generate` to build the reference contracts too and this looks ready.

**Comment by ssoroka** [comment]
@glinton am I good to merge?

**Review by glinton — APPROVED** [review]

**Pull request merged**